sdk/generic-sdk-core/generic-core/src/main/java/com/generic/core/http/policy/RequestRetryCondition.java

```diff
@@ -0,0 +1,76 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.generic.core.http.policy;
+
+import com.generic.core.http.models.HttpResponse;
+
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * Information about the request that failed, used to determine whether a retry should be attempted.
+ */
+public final class RequestRetryCondition {
+    private final HttpResponse response;
+    private final Throwable throwable;
+    private final int tryCount;
+    private final List<Throwable> retriedExceptions;
+
+    /**
+     * Creates a new ShouldRetryInfo object.
+     *
+     * @param response The HTTP response of the request that failed.
+     * @param throwable The throwable of the request that failed.
+     * @param tryCount The number of tries that have been attempted.
+     * @param retriedExceptions The list of exceptions that have been encountered during retries.
+     */
+    RequestRetryCondition(HttpResponse response, Throwable throwable, int tryCount,
+                          List<Exception> retriedExceptions) {
+        this.response = response;
+        this.throwable = throwable;
+        this.tryCount = tryCount;
+        this.retriedExceptions = retriedExceptions == null
+            ? Collections.emptyList() : Collections.unmodifiableList(retriedExceptions);
+    }
+
+    /**
+     * Gets the HTTP response of the request that failed.
+     * <p>
+     * This may be null if the request failed with a throwable and no response was received.
+     *
+     * @return The HTTP response of the request that failed.
+     */
+    public HttpResponse getResponse() {
+        return response;
+    }
+
+    /**
+     * Gets the throwable of the request that failed.
+     * <p>
+     * This may be null if the request failed with a response and no throwable was received.
+     *
+     * @return The throwable of the request that failed.
+     */
+    public Throwable getThrowable() {
```

Comment:
I'd make this `Exception` as well and change the API names to be `Exception` instead of `Throwable`.